### DISCUSSION TITLE
My staff is not recognized as a staff during GRID

### DISCUSSION THREAD

**Discussion by fovea1959** [Audiveris/audiveris#739] [General]
I have a nice clean PDF (it's actually output from Lilypond) that I am learning to use Audiveris with.

[AllOfMe.pdf](https://github.com/user-attachments/files/16103166/AllOfMe.pdf)

The 6th staff (starting with measure 17) is just not recognized as a staff. The problem seems to occur during GRID, though BINARY *is* rendering the horizontal lines in the problem staff at different thicknesses.

I read the parts of fixing staff problem, but those seem oriented towards fixing short staves, not adding one.

I am running on Linux, did a build from source against commit a721c5338038f6eda1952b33e9580c6766e68aef (tag 5.2.1)

**Comment by hbitteur** [comment]
I just tried your example using the development branch (tag 5.4-alpha).
All staves are recognized, including the very first staff which is void, musically speaking (it's just a title in fact).

I checked the thickness of the different staff lines: always 2 pixels.
Since the input is a PDF, and we now use a different PDF reader (PdfBox instead of PdfRenderer), this might be the root cause. My advice would thus be to upgrade to 5.4-alpha.

Also, in the following step (HEADERS), the engine complains about missing header clef in all staves except the second staff. I'm afraid the OMR engine is right and your input score is wrong! You should ask Lilypond to export legal staves... :-)

**Comment by hbitteur** [comment]
An additional suggestion: Ask Lilypond to use a more common text font.
The one in use looks like Finale Jazz Text, where even lowercase letters appear in capitals. This leads to very poor recognition by OCR.

**Reply by fovea1959** [reply]
I know that font is not the best for OMR, but that file is pretty representative of what I'll be working with when I am done testing.

**Comment by fovea1959** [comment]
development did not have the issue. It did much better with the music, and pretty much tanked on the lyrics and chords, but that's ok.
